src/instream.rs

```diff
@@ -370,7 +396,7 @@ impl<'a> InStreamReader<'a> {
 	// TODO: To acheive speed *and* safety I can use iterators. That will be in a future API.
 }
 
-impl<'a> Drop for InStreamReader<'a> {
+/*impl<'a> Drop for InStreamReader<'a> {
```

Comment:
Whats the reason for not doing this in `Drop`? Isn't it more unreliable to hope that the user remembers to call `end_read()`?

Also I assume you meant to delete this, as with `OutStreamWriter` rather than just comment it out?

Reply:
Ah, yeah, was meant to delete. Initially commented it out when I was testing.

I'll need to try what soundio thinks about calling `end_read/write` twice. If it doesn't like that being done, I suppose it's possible to set a flag when `start_read/write` has been called and only call `end_read/write` on drop if not yet called.